drivers/clock_control/clock_control_mspm0.c

```diff
@@ -197,15 +198,14 @@ static int clock_mspm0_init(const struct device *dev)
 #endif
 #endif
 
-#if MSPM0_LFCLK_ENABLED
 #if DT_SAME_NODE(DT_LFCLK_CLOCKS_CTRL, DT_NODELABEL(lfxt))
 	DL_SYSCTL_LFCLKConfig config = {0};
 
 	DL_SYSCTL_setLFCLKSourceLFXT(&config);
```

Comment:
DL_SYSCTL_setLFCLKSourceLFXT is unavailable on MSPM0 - C series which leads to build error.. 
Please introduced a guard.